crates/fulf/src/interface/mod.rs

```diff
@@ -0,0 +1,637 @@
+    ///
+    /// # Alternatives
+    ///
+    /// If you need a better control over algorithms, rules and directory
+    /// traversal, use `setter` function.
+    ///
+    /// If you need to read files in a manner different from `ignore::Walk`,
+    /// you can use `spawner` function.
+    ///
+    /// If you need something much different than anything there,
+    /// go and write it yourself.
+    #[inline]
+    pub fn default_searcher(
+        path: impl AsRef<Path>,
+        needle: impl Into<Box<str>>,
+        sort_and_print: impl FnMut(&mut [MWP], usize),
+    ) -> Option<(Vec<MWP>, usize)> {
+        with_fzy_algo(path, needle, 1024_usize.next_power_of_two(), sort_and_print)
+    }
+
+    /// A function to use default fuzzy-search algorithm.
+    ///
+    /// # Returns
+    ///
+    /// Return `None` if the root path cannot be represented as a utf8.
+    ///
+    /// # Maximum line length
+    ///
+    /// `max_line_len` sets maximum number of bytes for any line.
+    ///
+    /// If the line exceeds that number, it is not checked for match at all.
+    ///
+    /// Reasons:
+    ///
+    /// The speed of line-fuzzing is non-linear, thus lines too big
+    /// can slow down the task significantly. And there's very few reasons
+    /// for a line to exceed, for example, 1024 bytes:
+    ///
+    /// 1. This is a line in a text that is not code.
+    ///
+    /// 2. This is a non-formatted line of automatically generated code.
+    ///
+    /// 3. This is a very bad code.
+    ///
+    /// 4. Some very rare other reasons, like giant right-shifted branching.
+    ///
+    /// And in any of those cases there's probably no point in fuzzing such line.
+    #[inline]
+    pub fn with_fzy_algo(
+        path: impl AsRef<Path>,
+
+        needle: impl Into<Box<str>>,
+        max_line_len: usize,
+
+        sort_and_print: impl FnMut(&mut [MWP], usize),
+    ) -> Option<(Vec<MWP>, usize)> {
+        let needle = needle.into();
+
+        if needle.is_empty() || needle.len() > max_line_len {
+            return Default::default();
+        }
+
+        let r = {
+            let mut r = Rules::new();
+            r.bonus_threads = if cfg!(target_pointer_width = "64") {
+                2
+            } else {
+                0
+            };
+
+            r
+        };
+
+        let path = path.as_ref();
+        let dir_iter = ignore::Walk::new(path);
+
+        let root_folder = path.to_str()?;
+
+        let is_ascii = needle.is_ascii();
+
+        Some(if is_ascii {
+            // ascii
+            let fzydata = FzyData::new(root_folder.into(), needle, max_line_len);
+            FzyAscii::setter(dir_iter, fzydata, r, sort_and_print)
+        } else {
+            // utf8
+            let algo = move |needle: &str, line: &str| {
+                if line.len() > max_line_len {
+                    None
+                } else {
+                    utf8::match_and_score_with_positions(needle, line)
+                }
+            };
```

Comment:
Oops! There's a mistake: I made the the rg-like lines, but completely forgot to drop the filename from the scored string. I'll fix it as soon as possible.

Btw, I can just give the length of the partial path right to the algorithm, so there's no need in regex evaluation. Also, I can trim the whitespaces from the start of the line and add the trimmed number to the column number. Probably, should not trim all whitespaces, because it would change the picture of a line?
I dunno whether trimming all the spaces from the start or leaving up to 2-4 spaces would be better, need other opinion there.

Reply:
> can trim the whitespaces from the start of the line and add the trimmed number to the column number.

Actually I want this before! The leading whitespaces are meaningless, I think it's fine to trim the leading and trailing whitespaces.

Reply:
> > can trim the whitespaces from the start of the line and add the trimmed number to the column number.
> 
> Actually I want this before! The leading whitespaces are meaningless, I think it's fine to trim the leading and trailing whitespaces.

It's there: aeabfe1.